server/core/runtime/pre_workflow_hook_runner.go

```diff
@@ -47,8 +54,25 @@ func (wh DefaultPreWorkflowHookRunner) Run(ctx models.WorkflowHookCommandContext
 	if err != nil {
 		err = fmt.Errorf("%s: running %q in %q: \n%s", err, command, path, out)
 		ctx.Log.Debug("error: %s", err)
-		return "", err
+		return "", "", err
+	}
+
+	// Read the value from the "outputFilePath" file
```

Comment:
This is a similar block to the post workflow function. Could we create a function that does the same logic and use it in both places?

Reply:
Not unless we created a generic workflow_hook_runner interface to define functions for pre/post to implement, or we refactor them into one class.

Reply:
I meant a single function in a `utils.go` type of package to use the same block on line 60 in this file and the `post` workflow hook runner. I did not mean to update the interface for helper functions as that would be a much more involved refactor.

Reply:
We could add this to the `utils.go` but as you noticed post and pre workflow hooks are basically the same with different names, I believe a major refactor would be the better way to go and I'm willing to do this, if you wish I can start this on this PR or I can open another PR as you [suggested here](https://github.com/runatlantis/atlantis/pull/2441#issuecomment-1351694854).

Reply:
Ok that's fair enough. Let's refactor in the next pr